### DISCUSSION TITLE
Get plugin config from another plugin

### DISCUSSION THREAD

**Discussion by cosmy81** [saleor/saleor#10432] [Q&A]
Hi all.
I'm writing a new plugin that should make use of mail. I'd like to use the mail server configuration stored in user_email plugin just not to create too many places to get updated.
How can I get the configuration of another plugin? If I initialize the plugin I get the default configuration, so i think there should be used a method to get config data from database.
Thank you in advance

**Comment by korycins** [comment]
Hey @cosmy81,

Make sure that you can't use UserEmailPlugin first. The implementation of this is generic so if you just want to connect to the email provider and send email from the template, then it should work. 

We don't have any interface to fetch other's plugin configuration as we want to keep plugins isolated between each other.

Wouldn't be better, to turn off the UserEmailPlugin (or fully remove it from saleor/settings.py::PLUGINS), and use `UserEmailPlugin` code to built custom plugin that will provide a required configuration?

**Reply by cosmy81** [reply]
Yes it could be a good solution but the problem is still the same: if I use another plugin to share mail configuration for everywhere... i can't access it outside the plugin. 
There's should a way to use a plugin config, I mean, when a plugin is initialized inside the code it comes with the right config stored in the db, not the default one.

**Reply by timuric** [reply]
@cosmy81 sharing config between plugins would make them dependent on each other, which is not desirable

**Reply by cosmy81** [reply]
It depends. I think that replicating mail server connection config in every plugin that makes use of mail is a worst practice. 
On the other hand having a place in the dashboard where the mail config can be changed is good. At the moment a plugin is the only solution to do that.

**Reply by carlosa54** [reply]
@cosmy81 depending on how you implement the custom `UserEmailPlugin` you can make it as generic as you want and if you implement other plugins that you want to send emails you can do so as follows inside the other plugin

```
manager = get_plugins_manager()
manager.notify(...)
``` 

The above will iterate over the active plugins that implements `notify` and it will find your custom `UserEmailPlugin` solving your initial problem of sharing mail configuration. In this case your custom `UserEmailPlugin` will be the global interface for sending emails through plugins or other parts of the codebase
